File: backend/models/link.js
import mongoose from "mongoose";

const linkSchema = new mongoose.Schema(
  {
    title: { type: String, required: true, trim: true },
    url: { type: String, required: true, trim: true },
    category: {
      type: String,
      enum: ["Academic", "Hostel", "Club", "Other"],
      default: "Other",
    },
    iconURI: { type: String, trim: true },
  },
  { timestamps: true }
);

const Link = mongoose.model("Link", linkSchema);
export default Link;

File: frontend/lib/models/link.dart
class Link {
  final String id;
  final String title;
  final String url;
  final String category;
  final String? iconURI;

  Link({
    required this.id,
    required this.title,
    required this.url,
    required this.category,
    this.iconURI,
  });

  factory Link.fromJson(Map<String, dynamic> json) {
    return Link(
      id: json['_id'],
      title: json['title'],
      url: json['url'],
      category: json['category'],
      iconURI: json['iconURI'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'url': url,
      'category': category,
      'iconURI': iconURI,
    };
  }
}
